<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Statistics for Biologists: An Introduction to the application, understanding, and philosophy of statistical analysis in life science</title>
  <meta name="description" content="Statistics for Biologists: An Introduction to the application, understanding, and philosophy of statistical analysis in life science" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Statistics for Biologists: An Introduction to the application, understanding, and philosophy of statistical analysis in life science" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Statistics for Biologists: An Introduction to the application, understanding, and philosophy of statistical analysis in life science" />
  
  
  

<meta name="author" content="Alexander Kaier" />


<meta name="date" content="2021-05-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="descriptive-data-analysis.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Statistics for Biologists</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#why-is-statistics-in-life-science-important"><i class="fa fa-check"></i>Why is statistics in life science important?</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-to-expect-from-this-course"><i class="fa fa-check"></i>What to expect from this course</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-not-to-expect-from-this-course"><i class="fa fa-check"></i>What not to expect from this course</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#no-jack-of-all-trades"><i class="fa fa-check"></i>No Jack of all trades</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#an-ode-to-r"><i class="fa fa-check"></i>An ode to R</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#roadmap"><i class="fa fa-check"></i>Roadmap</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="descriptive-data-analysis.html"><a href="descriptive-data-analysis.html"><i class="fa fa-check"></i><b>1</b> Descriptive data analysis</a></li>
<li class="chapter" data-level="2" data-path="distributions.html"><a href="distributions.html"><i class="fa fa-check"></i><b>2</b> Distributions</a>
<ul>
<li class="chapter" data-level="2.1" data-path="distributions.html"><a href="distributions.html#what-are-probability-distributions"><i class="fa fa-check"></i><b>2.1</b> What are probability distributions</a></li>
<li class="chapter" data-level="2.2" data-path="distributions.html"><a href="distributions.html#what-do-we-need-distributions-for"><i class="fa fa-check"></i><b>2.2</b> What do we need distributions for</a></li>
<li class="chapter" data-level="2.3" data-path="distributions.html"><a href="distributions.html#distributions-in-r"><i class="fa fa-check"></i><b>2.3</b> Distributions in R:</a></li>
<li class="chapter" data-level="2.4" data-path="distributions.html"><a href="distributions.html#continuous-probabality-distributions"><i class="fa fa-check"></i><b>2.4</b> Continuous probabality distributions</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="distributions.html"><a href="distributions.html#uniform-distribution"><i class="fa fa-check"></i><b>2.4.1</b> Uniform distribution</a></li>
<li class="chapter" data-level="2.4.2" data-path="distributions.html"><a href="distributions.html#normal-distribution"><i class="fa fa-check"></i><b>2.4.2</b> Normal distribution</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="distributions.html"><a href="distributions.html#discrete-distributions"><i class="fa fa-check"></i><b>2.5</b> Discrete distributions</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="distributions.html"><a href="distributions.html#bernouille-distribution"><i class="fa fa-check"></i><b>2.5.1</b> Bernouille distribution</a></li>
<li class="chapter" data-level="2.5.2" data-path="distributions.html"><a href="distributions.html#binomial-distribution"><i class="fa fa-check"></i><b>2.5.2</b> Binomial distribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-transformation.html"><a href="data-transformation.html"><i class="fa fa-check"></i><b>3</b> Data transformation</a></li>
<li class="chapter" data-level="4" data-path="reject-or-not-to-reject.html"><a href="reject-or-not-to-reject.html"><i class="fa fa-check"></i><b>4</b> Reject or not to reject</a>
<ul>
<li class="chapter" data-level="4.1" data-path="reject-or-not-to-reject.html"><a href="reject-or-not-to-reject.html#t-test"><i class="fa fa-check"></i><b>4.1</b> t-test</a></li>
<li class="chapter" data-level="4.2" data-path="reject-or-not-to-reject.html"><a href="reject-or-not-to-reject.html#f-test"><i class="fa fa-check"></i><b>4.2</b> F-test</a></li>
<li class="chapter" data-level="4.3" data-path="reject-or-not-to-reject.html"><a href="reject-or-not-to-reject.html#chi2-test"><i class="fa fa-check"></i><b>4.3</b> <span class="math inline">\(\chi^2\)</span>-test</a></li>
<li class="chapter" data-level="4.4" data-path="reject-or-not-to-reject.html"><a href="reject-or-not-to-reject.html#one-test-to-rule-them-all-the-likelihood-ratio-test"><i class="fa fa-check"></i><b>4.4</b> One test to rule them all: The likelihood ratio test</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>5</b> Results</a></li>
<li class="chapter" data-level="6" data-path="interactive-component.html"><a href="interactive-component.html"><i class="fa fa-check"></i><b>6</b> Interactive component</a></li>
<li class="chapter" data-level="7" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>7</b> Conclusion</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="the-math-behind-the-model.html"><a href="the-math-behind-the-model.html"><i class="fa fa-check"></i><b>A</b> The math behind the model</a></li>
<li class="chapter" data-level="B" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>B</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistics for Biologists: An Introduction to the application, understanding, and philosophy of statistical analysis in life science</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Statistics for Biologists: An Introduction to the application, understanding, and philosophy of statistical analysis in life science</h1>
<p class="author"><em>Alexander Kaier</em></p>
<p class="date" style="margin-top: 1.5em;"><em>2021-05-10</em></p>
</div>
<div id="preface" class="section level1 unnumbered">
<h1>Preface</h1>
<div id="why-is-statistics-in-life-science-important" class="section level2 unnumbered">
<h2>Why is statistics in life science important?</h2>
<p>Fortunately for me (and at this point maybe unfortunately for some readers), there is virtually no single domain in the life sciences that has not been invaded by statistical analysis. And the reason for that is easy:
<strong>Statistics is the best and sometimes only way we have to gain insight from data.</strong></p>
<p>Tell a little bit about Fisher and other famous Biostatisticians.</p>
<p>Tell about the deadly pitfalls of falsely applied statistics (and mention John Ioannidis)</p>
</div>
<div id="what-to-expect-from-this-course" class="section level2 unnumbered">
<h2>What to expect from this course</h2>
<p>This course will teach you the foundations of statistics in life science as comprehensible as possible. The material will neither lean towards the mathematical theory of statistics, nor will it be purely applied, since both of these topics are comprehensive enough to fill multiple volumes on their own. I rather chose to go the golden middle path. Here, three major methods will alternate in their utilization and depth. These are: mathematical concepts, programming, and (most importantly) textual explanations of everything of importance for the concept or problem at hand to be understood thoroughly.</p>
<p>The mathematics will be used according to the rule “as much as necessary, as little as possible”. This is NOT a course about mathematical statistics, so don’t be afraid about to much maths. Nevertheless, statistics is a subdiscipline of mathematics after all, and therefore, there is no way to avoid mathematics altogether. But there is no reason to be discouraged, because all equations and where necessary the derivations thereof will be explained in great detail. After this course, you will be able to read and navigate through the basic mathematics of statistics like an experienced sailor can read the waves and stars to find their<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> way home.</p>
<p>The second method of teaching used in this course, the programming, is as essential as the mathematics. Although it is called programming, we will mainly focus on the analysis of data by means of the programming language R (depending on your career, programming usually encompasses much more than mere data analysis). It might take some time until you get used to using R for data analysis, because I expect most readers to come from a background where data analysis was performed in one of the many available spreadsheet programs (e.g. Excel). Such kind of software is also called WYSIWYG (what you see is what you get) because it works with buttons and widgets to click on and drag around. Although visually appealing, using a graphical user interface (GUI) for data analysis is actually hindering compared to code based software like R. I too came from this background and had to get used to this kind of analysis software. At this point I can give you the spoiler that all efforts learning code based data analysis are more than repaid once you get stuck with your graphical analysis software. For a more elaborate pleading in favor of R see <a href="index.html#an-ode-to-r">this section about the perks of R in data analysis</a>. In order to follow through the course you will have to have R installed on your machine (it works on all prevalent OS). Also, you will need some additional packages throughout the course. R packages are like libraries of code (and are often even called libraries) because they contain code written by other people in order to facilitate our analysis pipeline and focus on the fun part without having to meddle too much with the programming part itself. The code I use in this course will be embedded in blocks. Most of the time, such a block is subsided by its output, like in the following example:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting a random seed reproducible simulation</span></span>
<span id="cb1-2"><a href="index.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb1-3"><a href="index.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Defining the predictor variable (uniformly distributed)</span></span>
<span id="cb1-4"><a href="index.html#cb1-4" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">runif</span>(<span class="at">n=</span><span class="dv">100</span>, <span class="at">min=</span><span class="dv">0</span>, <span class="at">max=</span><span class="dv">3</span>)</span>
<span id="cb1-5"><a href="index.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Defining the response variable (normally distributed)</span></span>
<span id="cb1-6"><a href="index.html#cb1-6" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="at">n=</span><span class="dv">100</span>, <span class="at">mean=</span><span class="dv">2</span><span class="sc">*</span>x, <span class="at">sd=</span><span class="fl">0.5</span>)</span>
<span id="cb1-7"><a href="index.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a linear model from x and y</span></span>
<span id="cb1-8"><a href="index.html#cb1-8" aria-hidden="true" tabindex="-1"></a>lf <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> x)</span>
<span id="cb1-9"><a href="index.html#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting the model coefficients </span></span>
<span id="cb1-10"><a href="index.html#cb1-10" aria-hidden="true" tabindex="-1"></a>coefs <span class="ot">=</span> lf<span class="sc">$</span>coefficients</span>
<span id="cb1-11"><a href="index.html#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a scatter plot from x and y</span></span>
<span id="cb1-12"><a href="index.html#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(x,y)</span>
<span id="cb1-13"><a href="index.html#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw a line with parameters from the model</span></span>
<span id="cb1-14"><a href="index.html#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(coefs[<span class="dv">1</span>], coefs[<span class="dv">2</span>], <span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="finalproj_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># View summary statistics of the linear model</span></span>
<span id="cb2-2"><a href="index.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lf)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = y ~ x)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -1.11899 -0.30661 -0.00987  0.29817  1.10861 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept) -0.00448    0.09803  -0.046    0.964    
## x            1.98503    0.05697  34.843   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.4846 on 98 degrees of freedom
## Multiple R-squared:  0.9253, Adjusted R-squared:  0.9245 
## F-statistic:  1214 on 1 and 98 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>In the preceding example, there are two blocks of code, and two outputs. The first block generates some data, fits a linear model using these data, and plots the data and a trend line, while the second block prints the summary of the linear model. The output are the model parameters and statistics. Don’t worry if you don’t understand the code or the output for now, we will come to it at an appropriate time. All code in this course is explained on the run so that you never get the feeling of being lost in it. The example above is only there to show how it will look like. Throughout this course, I encourage you to copy the code presented, and execute it on your own machine. Alter it, change the parameters, change the data, try different things with it. There is hardly a scientific discipline where practice leads to a steeper learning curve and a better understanding of the subject than in coding (mathematics and physics may be at a similar level). In the worst case, your code breaks and you check what went wrong. Learning something in the troubleshooting process is a valuable lesson and once you start code based data analysis you will encounter errors on a regular basis. More often than not, the error is just a typo or due to other simple reasons. You will learn how to read and interpret error messages like the morning newspaper. Believe me, the rate at which you are confronted with errors does not really change with gained expertise. What changes is your attitude regarding the troubleshooting process.</p>
<p>The last pillar of the didactics used in this course is the actual text, where I explain what we do, how we do it, and most importantly, <em>why</em> we do it. When learning something, I always favor a well-told narrative of the subject over a sterile and indifferent way of conveying it. Although science and statistics ought to be neutral and unbiased, this should not be confused with boring, and the teaching of these subjects can and should be done in a passionate and animating way. For that reason, where appropriate, I will not only talk about the theory of a subject itself, but also to some extent about its impact on society, its use and misuse in science and politics, or about the bright minds behind the concept and their reason of deduction. After all, no topic is isolated and its beauty is best seen when put into context and with regards to the interdisciplinary ecosystem it is embedded in. We will for example see that some of the most famous statistical concepts were derived from people working in the biological sciences who tried to evaluate their observations. I beg your pardon if for these reasons the course might appear somewhat inflated but I think it is better to write three sentences too much than missing an important one, thereby sending the reader on a frustrating, day-long quest through the realms of the internet in order to find an explanation for the topic at hand (although also such an odyssey can contribute to your understanding).</p>
<p>Each chapter will be accompanied by some exercises. These are categorized into <em>easy</em>, <em>medium</em>, and <em>difficult</em>. They will test your understanding of the mathematical theory, demand computational analysis, data visualization, and predictions by programming your way through the problem. Finally, there will be some questions testing your general understanding and statistical intuition that you will have hopefully developed up to this point of the course. Don’t refrain if you cannot solve some of the problems, as long as you gain the impression that you understand the course material well enough to make rational decisions regarding the analysis of your own data.</p>
<!--
One final treat I will present to you in this course is owed to me being obsessed with interactive data charts. Publishing online has the special benefit that I can include graphical representations of important statistical concepts which are furnished with control elements. You can tweak the parameters within the charts and experience in real time, how the concept at hand changes its behavior qualitatively and quantitatively. I think this is an exciting and fruitful way of conveying science and a method of transporting ideas and visions from someone's mind to the eye of the reader. Please be aware that you need a stable internet connection in order to interact with these kind of charts.
-->
<div id="htmlwidget-0349e08474188736d854" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0349e08474188736d854">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150"],[5.1,4.9,4.7,4.6,5,5.4,4.6,5,4.4,4.9,5.4,4.8,4.8,4.3,5.8,5.7,5.4,5.1,5.7,5.1,5.4,5.1,4.6,5.1,4.8,5,5,5.2,5.2,4.7,4.8,5.4,5.2,5.5,4.9,5,5.5,4.9,4.4,5.1,5,4.5,4.4,5,5.1,4.8,5.1,4.6,5.3,5,7,6.4,6.9,5.5,6.5,5.7,6.3,4.9,6.6,5.2,5,5.9,6,6.1,5.6,6.7,5.6,5.8,6.2,5.6,5.9,6.1,6.3,6.1,6.4,6.6,6.8,6.7,6,5.7,5.5,5.5,5.8,6,5.4,6,6.7,6.3,5.6,5.5,5.5,6.1,5.8,5,5.6,5.7,5.7,6.2,5.1,5.7,6.3,5.8,7.1,6.3,6.5,7.6,4.9,7.3,6.7,7.2,6.5,6.4,6.8,5.7,5.8,6.4,6.5,7.7,7.7,6,6.9,5.6,7.7,6.3,6.7,7.2,6.2,6.1,6.4,7.2,7.4,7.9,6.4,6.3,6.1,7.7,6.3,6.4,6,6.9,6.7,6.9,5.8,6.8,6.7,6.7,6.3,6.5,6.2,5.9],[3.5,3,3.2,3.1,3.6,3.9,3.4,3.4,2.9,3.1,3.7,3.4,3,3,4,4.4,3.9,3.5,3.8,3.8,3.4,3.7,3.6,3.3,3.4,3,3.4,3.5,3.4,3.2,3.1,3.4,4.1,4.2,3.1,3.2,3.5,3.6,3,3.4,3.5,2.3,3.2,3.5,3.8,3,3.8,3.2,3.7,3.3,3.2,3.2,3.1,2.3,2.8,2.8,3.3,2.4,2.9,2.7,2,3,2.2,2.9,2.9,3.1,3,2.7,2.2,2.5,3.2,2.8,2.5,2.8,2.9,3,2.8,3,2.9,2.6,2.4,2.4,2.7,2.7,3,3.4,3.1,2.3,3,2.5,2.6,3,2.6,2.3,2.7,3,2.9,2.9,2.5,2.8,3.3,2.7,3,2.9,3,3,2.5,2.9,2.5,3.6,3.2,2.7,3,2.5,2.8,3.2,3,3.8,2.6,2.2,3.2,2.8,2.8,2.7,3.3,3.2,2.8,3,2.8,3,2.8,3.8,2.8,2.8,2.6,3,3.4,3.1,3,3.1,3.1,3.1,2.7,3.2,3.3,3,2.5,3,3.4,3],[1.4,1.4,1.3,1.5,1.4,1.7,1.4,1.5,1.4,1.5,1.5,1.6,1.4,1.1,1.2,1.5,1.3,1.4,1.7,1.5,1.7,1.5,1,1.7,1.9,1.6,1.6,1.5,1.4,1.6,1.6,1.5,1.5,1.4,1.5,1.2,1.3,1.4,1.3,1.5,1.3,1.3,1.3,1.6,1.9,1.4,1.6,1.4,1.5,1.4,4.7,4.5,4.9,4,4.6,4.5,4.7,3.3,4.6,3.9,3.5,4.2,4,4.7,3.6,4.4,4.5,4.1,4.5,3.9,4.8,4,4.9,4.7,4.3,4.4,4.8,5,4.5,3.5,3.8,3.7,3.9,5.1,4.5,4.5,4.7,4.4,4.1,4,4.4,4.6,4,3.3,4.2,4.2,4.2,4.3,3,4.1,6,5.1,5.9,5.6,5.8,6.6,4.5,6.3,5.8,6.1,5.1,5.3,5.5,5,5.1,5.3,5.5,6.7,6.9,5,5.7,4.9,6.7,4.9,5.7,6,4.8,4.9,5.6,5.8,6.1,6.4,5.6,5.1,5.6,6.1,5.6,5.5,4.8,5.4,5.6,5.1,5.1,5.9,5.7,5.2,5,5.2,5.4,5.1],[0.2,0.2,0.2,0.2,0.2,0.4,0.3,0.2,0.2,0.1,0.2,0.2,0.1,0.1,0.2,0.4,0.4,0.3,0.3,0.3,0.2,0.4,0.2,0.5,0.2,0.2,0.4,0.2,0.2,0.2,0.2,0.4,0.1,0.2,0.2,0.2,0.2,0.1,0.2,0.2,0.3,0.3,0.2,0.6,0.4,0.3,0.2,0.2,0.2,0.2,1.4,1.5,1.5,1.3,1.5,1.3,1.6,1,1.3,1.4,1,1.5,1,1.4,1.3,1.4,1.5,1,1.5,1.1,1.8,1.3,1.5,1.2,1.3,1.4,1.4,1.7,1.5,1,1.1,1,1.2,1.6,1.5,1.6,1.5,1.3,1.3,1.3,1.2,1.4,1.2,1,1.3,1.2,1.3,1.3,1.1,1.3,2.5,1.9,2.1,1.8,2.2,2.1,1.7,1.8,1.8,2.5,2,1.9,2.1,2,2.4,2.3,1.8,2.2,2.3,1.5,2.3,2,2,1.8,2.1,1.8,1.8,1.8,2.1,1.6,1.9,2,2.2,1.5,1.4,2.3,2.4,1.8,1.8,2.1,2.4,2.3,1.9,2.3,2.5,2.3,1.9,2,2.3,1.8],["setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","setosa","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","versicolor","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica","virginica"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sepal.Length<\/th>\n      <th>Sepal.Width<\/th>\n      <th>Petal.Length<\/th>\n      <th>Petal.Width<\/th>\n      <th>Species<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>To sum it up, after finishing this course, you will feel confident around different types of data, and recognize potential pitfalls in the design of experimental studies. You will not start sweating in the face of the myriads of options to analyze your data and be able to distinct different types of statistical tests and models. In short, <strong>you will have learned how to think statistically</strong>.</p>
</div>
<div id="what-not-to-expect-from-this-course" class="section level2 unnumbered">
<h2>What not to expect from this course</h2>
<p>All efforts in teaching any scientific topic are in vain if the reader is not willing to put effort into the understanding of the taught subjects. I will try my best to convey the information of this tutorial in an intuitive and understandable way. Eventually however, it is up to you, the reader, to think about the material, play around with the code examples, give the exercises a shot and, most important, <strong>transform knowledge to insight</strong>. In the end, you should be able to apply the learned material to your own research (or approach the things you do daily from a statistical point of view). I will not lie, this demands a certain amount of time and discipline. After all, Rome wasn’t built in one day. The result, however, is more than rewarding. As a biologist by training myself (I pursued my undergraduate studies in the field of synthetic biology), I sympathize with anyone who encounters problems with leaving their comfort zone and entering the realms of mathematical reasoning, understanding and dissecting equations, and statistical programming. However, the target groups of this tutorial are scientists and students who aim to become scientists in the not-so-far future. From such an audience I think it is legitimate to expect general curiosity about learning new things and especially those that make your life easier on the long run and will eventually make you a better scientist.</p>
</div>
<div id="no-jack-of-all-trades" class="section level2 unnumbered">
<h2>No Jack of all trades</h2>
<p>Although this tutorial aims to be quite comprehensive, I am totally aware of the fact that no single person can learn and know everything, according to the old saying “<em>Jack of all trades, master of none</em>”. After my undergraduate studies in biology I chose to pursue my graduate studies in the fields of theoretical biology and bioinformatics, with specialization on the analysis of genomic data. By doing so, I was not able to stay up to date with all the biological methods that are used in the wet lab. So in order to perform sound science, I have to rely on the expertise of wet lab scientists to actually design and conduct the experiments. I have the greatest respect towards all biologists and other wet lab scientists who struggle with the obstacles of their research, like maintaining their samples, dealing with suboptimal environmental conditions, tedious purification steps, time management for complex experiments and all the other things I do not know much about. However, as a bioinformatician it is my duty to know biology at least to some extent. Domain knowledge for any data analyst is paramount to select the right models to analyze the data at hand or answer a specific scientific question. As we will see later there is sometimes no single best way to analyze your data and the precise analysis can depend on the context given. But this kind of expertise is no one-way road.</p>
<p><strong>MENTION HERE THE IMPORTANCE OF THE FIRST POINT</strong>.</p>
<p>For this simple reason it is not a choice, but a necessity for any scientist in the life sciences, to know at least the foundations of statistics (and yes dear reader, these foundations comprise more than a simple t-test command). And while this tutorial aims at giving you exactly this foundations at hand, it can by no means cover all fields of modern statistics and analysis of biological data. However, this is not expected from any experimental scientist. It is very likely that at some point in your career you will encounter or generate data that will surpass your skills in data analysis. This is completely normal and whole professions like the biostatistician and the bioinformatician have emerged in order to analyze these data (these professions are of great demand nowadays due to the vast amount of data being generated by high-throughput sequencing in recent years). But it is still your job to perform rudimentary data analysis with small to moderate sized data sets independently. Also you should be in the position to find a common language with the biostatistician/bioinformatician at your department, when discussing experimental design and analysis. This fruitful conversations and consequently successful projects can only emerge if you know statistics to some extent, like the analyst knows the biology and principles of wet lab methods to some degree.</p>
</div>
<div id="an-ode-to-r" class="section level2 unnumbered">
<h2>An ode to R</h2>
<p>I first encountered the statistical programming language R when I had to accomplish the data analysis for the data I collected for my Bachelor thesis. Back then I had no idea about programming, R, or applied statistics (besides maybe the t-test, how much of your undergraduate statistics course do you remember and apply in your daily work?). Back then my supervisor forced me to do the data analysis using R and RStudio<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. Although a medical biologist by training, he loved programming, statistics and data analysis. I shall thank him forever for opening that door for me (he actually rather pushed me through it for my own good). As soon as I started doing my data analysis using R and RStudio, I realized the treasure I had discovered. During the remainder of this section, I will share with you the perks of R and RStudio that I learned appreciating over time.</p>
<p>The first and probably biggest advantage of R or any other programming language over spreadsheet software like Excel is that the analysis using R is self-documenting. What I mean by that is when you do your data analysis in R, you usually create an empty script first, which you then fill with the code that you need to load your data, analyze it at will and create and save output statistics and plots. This fundamentally different approach compared to spreadsheet software allows you to come back to your analysis at any time and break it down bit-by-bit. Imagine you get some experimental data and need to extract some summary statistics from them. After some time, you realize you made a mistake and have to redo the analysis. If you did it using some spreadsheet software, then good luck redoing all the clicks and drags and whatever steps you did in a reproducible way. If you used R instead, you just open the script you wrote for the analysis in RStudio, go through it line by line, change the erroneous part, and rerun the whole script. You virtually eliminate the potential for human errors, when doing the same thing twice. This also saves a huge amount of time, depending on the complexity of the analysis and the amount of data. If then anybody asks you what you precisely did, you can just hand them over the script or explain it to them like you would read a book. If you used some spreadsheet software instead, chances are high you wrote every single step down in your manual lab-book. If you are sufficiently meticulous in your documentation to have written it down so that you can actually redo a complex analysis, I bet writing down the whole analysis took you more time than you’d like to admit. This doubling of work almost never happens when using R. Also you are less prone to work with the raw data when using R. I have seen on multiple occacions now that some scientist hands me over a spreadsheet, where not only the raw data was included, but they also created already summary statistics and plots and whatever in the same spreadsheet. <strong>Never do that</strong>. Although I think of it as self-evident, I have seen this habit too often to not emphasize it here. When doing your data analysis, you load your data into R and then manipulate the version of the data in your running session<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>. Once you are done with the analysis, you save <em>a new document</em> with whatever insight you gained from your data. Never manipulate your raw data. This is only one of the many good habits you’ll take to you heart once you start working with data more professional and with R.</p>
<p>Second, some analyses can be <em>really</em> specific to the problem at hand. Chances are you search the internet for days in order to solve a specific problem or implement the solution as desired using some spreadsheet software. As the term <em>programming language</em> implies, R can be programmed, meaning that quite often it proves to be handy if you can create the desired solutions using a few lines of code which you then implement in your analysis. Many times when you want to use a specific program to analyze your data, this program expects a very specific way of organization within you dataset. If you can’t find someone who did the exact same data transformation before and published a program that you can use you might run into trouble. If you are able to transform your data yourself using some basic programming, there are no strings attached to you<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>. However, this is no book about programming, and therefore we will not dive deep into the core concepts of programming languages, but focus on the code that is necessary to conduct sound data analysis and statistics. But keep in mind the power you gain if you can solve a huge amount of problems that would otherwise prevent you from doing what you want to do using only a few lines of code.</p>
<p>Third, and related to the first point, sometimes you want to test different models and compare their fit to your data or create different summary statistics, based on the question to be answered. Using spreadsheet software (by now you probably guessed it already) it can take a lot of time to copy data, transform it, handle missing data appropriately and so on. Using R, these kind of tasks are usually accomplished by changing a few lines of code slightly, maybe by tweaking some function parameters or looping over different models to compare them. And again, the single steps are much more tractable if you do this using a programming based approach compared to spreadsheet software.</p>
<!--
Third, 

R has three great advantages over Excel and likewise:

- datasets that are too large for excel can still be analyzed.
- Open source for the win.
-->
</div>
<div id="roadmap" class="section level2 unnumbered">
<h2>Roadmap</h2>
<p><strong>Chapter 1</strong> will focus on <em>descriptive statistics</em>. Every experiment will at some point yield data that are to be analyzed <a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>. As the name already gives away, descriptive statistics aims at summarizing your data and putting it into context, like the kind of distribution your data follows. This topic will be covered at the very beginning for two reasons. First, it does not demand a large amount of statistical knowledge and can be applied directly after obtaining the data at hand. Therefore, you can directly see the fruits of your work without having to meddle to strong with statistical concepts and assumptions. Second, it is an important first step in further analysis. Since we can usually only obtain a (small) sample size compared to a much larger population, the aim of the overall statistical analysis is to infer certain properties of the population based on the sample at hand. But in order to do so, we first have to <em>describe</em> how the sample data looks like and behaves. Hence, we begin with descriptive statistics.</p>
<p><strong>Chapter 2</strong> will lead us to the concept of distributions. Although the term “distribution” is used frequently in everyday life, its meaning in stats lingo is not so well-known. We will explore the differences between data distributions and theoretical distribution. Also, you will learn why and how to apply the important concepts of <em>distribution parameters</em> and <em>parameter estimation</em>. Some important types of distributions will be presented and their characteristics illuminated. Also, we will see how to fit a distribution to the data at hand. The concepts learned here will form the foundations for the subsequent chapters and will prove invaluable for your further understanding of statistics.</p>
<p>In <strong>Chapter 3</strong>, we will dip our toes in the deep sea of regressions and explore the concept of <em>generalized linear models</em> (GLM). Don’t worry if this means nothing to you right now, it will all come at an appropriate time. This chapter is the logical consequence of the principles of distributions that we have learned before. The difference is that for GLM, the distribution parameter is no longer constant but depends on some variable within our data. In this way, we can explore the relations between two different data variables, the independent variable and the dependent variable. Although many people learning statistics do not connect these two concepts, this chapter paves the way for understanding statistical testing, as will be shown later.</p>
<p><strong>Chapter 4</strong> will deal with the concepts of correlation and covariance. These topics are in some way the next step after linear regression, because for the analysis of correlation and covariance, you need to lose the constraint of one variable depending on another, as is the case for linear regression. In correlation analysis, no dependency is assumed, and therefore only the tendency of different data variables to vary together is analyzed.</p>
<p><strong>Chapter 5</strong> will bring us finally to the concepts of statistical testing. Although some readers render this topic to be the most important one for their work and would therefore like to jump into it right away, there is a reason for putting it at the end of the course. Statistical testing unites all the knowledge and skills we have gained along the way. We will introduce the most important statistical tests and associate their theoretical background and results with the content of the preceding chapters. So I kindly ask you to not be hasty and enjoy the ride we take all the way down until this chapter in order to understand and appreciate it as much as possible.</p>
<p>Finally, there is the <strong>Appendix</strong>. Here, one part will be about <a href="the-math-behind-the-model.html#the-math-behind-the-model">the more advanced mathematics of the introduced topics</a>. Whenever I deem the mathematical details of a topic to be useful for a deeper understanding but out of scope and not necessary for this course, I will refer to this part. Here, the mathematics will be explained thoroughly and calculations will be done step-by-step, so that readers with non-mathematical background can follow along. The curious reader can then come here and try to retrace the mathematics that can lead to a deeper insight of the topic. However, this part is not necessary for understanding the content of this course and can be ignored at will.</p>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>Throughout this course, I will use the generic plural, even for a single person like in this case, where possible. If at some point the chosen examples do not reflect the reader’s identity it is to be mentioned that this is not intentional and should not be interpreted as such. The grammatical gender of a word does not depend on the identity or gender of the reader<a href="index.html#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>RStudio is what is called an <em>Integrated Development Environment</em>(IDE). This kind of software is there for helping you writing, testing, and managing code, and since R is a statistical programming language, R-Studio is particularly designed for data analysis tasks. If you have it not already, I strongly recommend downloading the version suitable for your OS under the <a href="https://www.rstudio.com/products/rstudio/download/">RStudio download page</a><a href="index.html#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>or if you work with spreadsheet software, create a copy and do the analysis using this copy<a href="index.html#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>Because of this precise feature of being programmable, in recent years hundreds of R packages were written specifically with the aim of doing biological data analysis. Most of these packages are collected and distributed using the package manager <a href="https://www.bioconductor.org/">Bioconductor</a>. Although you probably won’t need or use most of the packages provided there it can give you quite a nice impression of how versatile and powerful the programming paradigm is compared to other approaches for data analysis<a href="index.html#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>Although maybe unintuitive at first sight, we will stick with the plural use of the word ‘data’. The reason is twofold. First, the word ‘data’ is already the plural form of the singular word ‘datum’. Second, an experiment usually yields more than a single data point (or datum)<a href="index.html#fnref5" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="descriptive-data-analysis.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/alexanderkaier/statistics-for-biologists/edit/main/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/alexanderkaier/statistics-for-biologists/blob/main/index.Rmd",
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
